### PR TITLE
Test Results Component

### PR DESCRIPTION
**Pull request by meghanmc**
### Description: 
Closes issue #50 

This Component contains the necessary fields to replace the Test Result section of the green card:
greenCardTests

Start the app by running `npm run start` from `/pwa`
The form is rendered on the route `/tests`. Navigate to `http://localhost:3000/tests` to view.

### Notes:
Due to this being a shorter form, and not having shared attribute types in each section, I chose to write out each element of the form rather than use the JSON format (such as the one used for #84 ).

There are several 'TODO's that either require feedback from other team members or require functionality that is included in a separate PR.

### Screenshots (Optional): 
image

image

image

image

### Checklist:
- [ ] 90% Test Coverage

### CODE REVIEW

File: pwa/src/App.tsx [hunk 1]
```diff
@@ -35,14 +36,24 @@ const App: React.FunctionComponent = () => {
...
+                        >
+                            <DiagnosisForm />
+                        </Wrapper>
+                    </Route>
+                    <Route path="/tests">
+                        <Wrapper
+                            navTitle="Tests"
+                            // TODO: uncomment when Treatment component is completed
+                            // leftArrowText="Treatment"
+                            // leftArrowRoute="/treatment"
```
Comment (meghanmc) [inline pwa/src/App.tsx:52]: This should be done as part of either #104 or #103, depending on which is merged first.

File: pwa/src/components/TestResultsForm.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                                placeholder="0"
+                            >
+                                Protein
+                            </InputItem>
+                            <InputItem
+                                id="leukocytes-results-val"
+                                data-testid="leukocytes-results-val"
+                                placeholder="0"
+                            >
+                                Leukocytes
```
Comment (meghanmc) [inline pwa/src/components/TestResultsForm.tsx:126]: I attempted to determine what typical values for these tests are via Google, I found that it is likely either `+`, `-` or `[0-9]+`. We should check this with the existing database and update accordingly.

File: pwa/src/components/TestResultsForm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+})
+
+test('ua test results are rendered as number input boxes', () => {
+    const { getByTestId, queryByTestId } = render(<TestResultsForm />)
+    const uaResultFieldID = [
+        'glucose-results-val',
+        'nitrites-results-val',
+        'protein-results-val',
+        'leukocytes-results-val',
+    ]
```
Comment (abpwrs) [inline pwa/src/components/TestResultsForm.test.tsx:182]: could this list of fields be extracted out into its own data structure that both the main component and tests could share?

File: pwa/src/components/TestResultsForm.tsx (hunk 2 above)
Comment (abpwrs) [inline pwa/src/components/TestResultsForm.tsx:126]: duplicates could be made into for loop

pseudocode of idea:
```
for r in result_list:
<InputItem
    id="{r}-results-val"
    data-testid="{r}-results-val"
    placeholder="0"
> {r} </InputItem>
```

Comment (abpwrs) [reply pwa/src/components/TestResultsForm.tsx:126]: Then in the tests, you could just loop over the same structure instead of hard-coding in the same tags

Comment (meghanmc) [reply pwa/src/components/TestResultsForm.tsx]: added in this commit f4a2af5c04f38bfcd41a45343c6fea5ff5fb9872

File: pwa/src/components/TestResultsForm.test.tsx (hunk 3 above)
Comment (meghanmc) [reply pwa/src/components/TestResultsForm.test.tsx]: done in this commit f4a2af5c04f38bfcd41a45343c6fea5ff5fb9872

File: pwa/src/components/TestResultsForm.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                    <div
+                        id="blood-sugar-results"
+                        data-testid="blood-sugar-results"
+                    >
+                        <List renderHeader={() => 'Blood Sugar Results'}>
+                            <InputItem
+                                type="number"
+                                id="blood-sugar-results-val"
+                                data-testid="blood-sugar-results-val"
+                                placeholder="mg/dL" // TOOD: unsure if the unit is mg/dL or mmol/L, could leave this out
```
Comment (meghanmc) [inline pwa/src/components/TestResultsForm.tsx:84]: Unit for this field should be determined or this placeholder should be removed.

Comment (bmitchinson) [reply pwa/src/components/TestResultsForm.tsx:84]: I vote keeping it as mg/dl for now. Can be confirmed with a CHI rep at a later date.

Comment (meghanmc) [reply pwa/src/components/TestResultsForm.tsx:84]: Do you recommend leaving the TODO tag there or removing and creating an issue?

File: pwa/src/components/TestResultsForm.tsx (hunk 2 above)
Comment (meghanmc) [reply pwa/src/components/TestResultsForm.tsx]: After consulting the DB, it appears there is currently a mix of data.

tests_urine_glucose_result was large numbers (165 and 109 are some examples)
test_urinte_nitrites_result is a mix ('7.0', 'neg', 'pos', '+', and 'n' were some examples)
tests_urine_protein_result is also a mix ('6.5', '1', '30', 'neg', '+' were all given)
tests_urine_leukocytes_results is also a mix ('1', 'neg', 'pos', '+' all given)

The expected values/units (and corresponding DB cleaning) should be determined via consult with a CHI rep at a later date and updated accordingly. For now, they will allow any text input.

Comment (bmitchinson) [reply pwa/src/components/TestResultsForm.tsx]: 👍

File: pwa/src/components/TestResultsForm.tsx (hunk 4 above)
Comment (bmitchinson) [reply pwa/src/components/TestResultsForm.tsx:84]: Go ahead and make an issue, that's a good point.

Comment (meghanmc) [reply pwa/src/components/TestResultsForm.tsx]: Created #107

**Pull request merged**